src/Modal.php

```diff
@@ -29,176 +31,251 @@
-
-    private array $options = [];
-    private array $contentOptions = [];
-    private array $closeButtonOptions = [];
+    private const COLORS = ['is-primary', 'is-link', 'is-info', 'is-success', 'is-warning', 'is-danger', 'is-dark'];
+    private const SIZES = ['is-small', 'is-medium', 'is-large'];
+    private array $attributes = [];
+    private string $autoIdPrefix = 'w';
+    private array $closeButtonAttributes = [];
+    private string $closeButtonClass = 'modal-close';
     private string $closeButtonSize = '';
-    private bool $withoutCloseButton = true;
+    private array $contentAttributes = [];
+    private string $modalBackgroundClass = 'modal-background';
+    private string $modalButtonClass = 'button modal-button';
+    private string $modalClass = 'modal';
+    private string $modalContentClass = 'modal-content';
+    private array $toggleButtonAttributes = [];
     private string $toggleButtonLabel = 'Toggle button';
     private string $toggleButtonSize = '';
     private string $toggleButtonColor = '';
-    private array $toggleButtonOptions = [];
+    private bool $withoutCloseButton = true;
     private bool $withoutToggleButton = true;
 
     /**
-     * Returns a new instance with the specified main container options.
+     * The HTML attributes. The following special options are recognized.
+     *
+     * @param array $values Attribute values indexed by attribute names.
+     *
+     * @return self
+     *
+     * See {@see \Yiisoft\Html\Html::renderTagAttributes()} for details on how attributes are being rendered.
+     */
+    public function attributes(array $values): self
+    {
+        $new = clone $this;
+        $new->attributes = $values;
+        return $new;
+    }
+
+    /**
+     * Returns a new instance with the specified prefix to the automatically generated widget IDs.
+     *
+     * @param string $value The prefix to the automatically generated widget IDs.
+     *
+     * @return self
+     */
+    public function autoIdPrefix(string $value): self
+    {
+        $new = clone $this;
+        $new->autoIdPrefix = $value;
+        return $new;
+    }
+
+    /**
+     * Returns a new instance with the specified close button options.
      *
-     * @param array $value The main container options.
+     * @param array $value The close button options.
      *
      * {@see Html::renderTagAttributes()} for details on how attributes are being rendered.
      *
      * @return self
      */
-    public function options(array $value): self
+    public function closeButtonAttributes(array $value): self
     {
         $new = clone $this;
-        $new->options = $value;
+        $new->closeButtonAttributes = $value;
 
         return $new;
     }
 
     /**
-     * Returns a new instance with the specified toggle button label.
+     * Returns a new instance with the specified close button size.
      *
-     * @param string $value The toggle button label.
+     * @param string $value The close button size.
```

Comment:
Should mention what values are accepted. Also, maybe we can remove `is-` prefix?

Reply:
done.